parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -941,6 +993,37 @@ pub mod pallet {
 			Self::deposit_event(Event::ExecutionHeaderImported { block_hash, block_number });
 		}
 
+		fn get_mapping_bound() -> (u64, u64) {
+			<ExecutionHeadersOldestMapping<T>>::get().unwrap_or((1, 1))
+		}
+
+		fn update_mapping_bound(oldest: Option<u64>, latest: Option<u64>) {
+			let (previous_oldest, previous_latest) =
+				<ExecutionHeadersOldestMapping<T>>::get().unwrap_or((1, 1));
+			let new_oldest = oldest.unwrap_or(previous_oldest);
+			let new_latest = latest.unwrap_or(previous_latest);
+			<ExecutionHeadersOldestMapping<T>>::put((new_oldest, new_latest));
+		}
+
+		fn prune_older_execution_headers() {
+			let threshold = T::ExecutionHeadersPruneThreshold::get();
+			let stored_execution_headers = <ExecutionHeaders<T>>::count() as u64;
+
+			if stored_execution_headers > threshold {
+				let (mut oldest_mapping_to_delete, _) = Self::get_mapping_bound();
+				let execution_headers_to_delete =
+					stored_execution_headers.saturating_sub(threshold);
+				for _i in 0..execution_headers_to_delete {
+					let execution_header_hash =
+						<ExecutionHeadersMapping<T>>::get(oldest_mapping_to_delete);
+					<ExecutionHeadersMapping<T>>::remove(oldest_mapping_to_delete);
+					<ExecutionHeaders<T>>::remove(execution_header_hash);
+					oldest_mapping_to_delete += 1;
+				}
```

Comment:
https://github.com/Snowfork/snowbridge/pull/810/files#r1173313662

Reply:
while loop would be most likely run only 1 time. As it runs after every insert.